replicated_kv_store/client.go

```diff
@@ -7,13 +7,20 @@ import (
 	"log"
 	"net/http"
 	"reflect"
+	"time"
 
 	"github.com/krithikvaidya/distributed-dns/replicated_kv_store/protos"
 )
 
 //WriteCommand is called when the client sends the replica a write request.
 func (node *RaftNode) WriteCommand(operation []string, client string) bool {
 
+	for node.commitIndex != node.lastApplied {
```

Comment:
ApplyToStateMachine needs the full lock to apply the commits, but we aren't releasing the read lock in this loop. The loop body will have to be

```
for node.commitIndex != node.lastApplied {
        node.raft_node_mutex.RUnlock()
	time.Sleep(20 * time.Millisecond)
	node.raft_node_mutex.RLock()
}
```